File: python/bmi.py
def pobierzDane():
    m = float(input('Podaj masę ciała (kg): '))
    while m < 10 or m > 200:
        print('Błędna wartość')
        m = float(input('Podaj masę ciała (kg): '))

    w = float(input('Podaj wzrost (m): '))
    while w < 1 or w > 2.5:
        print('Błędna wartość')
        w = float(input('Podaj wzrost (m): '))

    return m, w


def obliczBmi(m, w):
    bmi = m / (w**2)
    return bmi


def wydrukujDiagnoze(bmi):
    if bmi < 18.5:
        print('niedowaga')
    elif bmi < 25:
        print('norma')
    elif bmi < 30:
        print('nadwaga')
    else:
        print('otyłość')


def main(args):
    m, w = pobierzDane()
    # print(m, w)
    bmi = obliczBmi(m, w)
    print(bmi)
    wydrukujDiagnoze(bmi)
    input('Naciśnij jakiś klawisz...')
    return 0


if __name__ == '__main__':
    import sys
    sys.exit(main(sys.argv))

File: cpp/bmi.cpp
/*
 * bmi.cpp
 */
using namespace std;

#include <iostream>

void pobierzDane(float &m, float &w) {
    cout << "Podaj masę ciała (kg): ";
    cin >> m;
    while (m < 10 || m > 200) {
        cout << "Błędna wartość" << endl;
        cout << "Podaj masę ciała (kg): ";
        cin >> m;
    }
    cout << "Podaj wzrost (m): ";
    cin >> w;
    while (w < 1 || w > 2.5) {
        cout << "Błędna wartość" << endl;
        cout << "Podaj wzrost (m): ";
        cin >> w;
    }
}

float obliczBmi(float m, float w) {
    float bmi = m / (w * w);
    return bmi;
}

void wydrukujDiagnoze(float bmi) {
    if (bmi < 18.5) {
        cout << "niedowaga";
    } else if (bmi < 25) {
        cout << "norma";
    } else if (bmi < 30) {
        cout << "nadwaga";
    } else {
        cout << "otyłość";
    }
}

int main(int argc, char **argv)
{
    // deklaracja zmiennych, może zawierać inicjację wartości zmiennej
    // deklarcja + inicjacja = definicja
    float m = 0;
    float w = 0;
    float bmi = 0;

    pobierzDane(m, w);
    bmi = obliczBmi(m, w);
    cout << bmi << endl;
    wydrukujDiagnoze(bmi);

	return 0;
}

